core/src/main/resources/lib/layout/breadcrumbs.js

```diff
@@ -22,15 +22,23 @@ window.breadcrumbs = (function () {
   var logger = function () {};
   // logger = function() { console.log.apply(console,arguments) };  // uncomment this line to enable logging
 
-  function makeMenuHtml(icon, iconXml, displayName) {
+  function makeMenuHtml(icon, iconXml, displayName, badge) {
     var displaynameSpan = "<span>" + displayName + "</span>";
+    const badgeSpan =
+      badge === null
+        ? ""
+        : `<span class="yui-menu-badge" tooltip="${badge.tooltip}">${badge.text}</span>`;
```

Comment:
Looks XSS-able. Badge tooltips are fairly likely to contain user-provided content. Please create the element via DOM API (or at least not via string concatenation / template string).

Reply:
<img width="1340" alt="XSS" src="https://user-images.githubusercontent.com/1831569/210324908-8e380f9c-cb7f-4f4e-a665-043138147469.png">

Reply:
could you provide steps to reproduce please?

Reply:
Write a plugin that declares a `ManagementLink` with user-defined `Badge` title and tooltip.

Use the XSS payloads `<img src=x onerror=alert(1)>` for the `title` and/or `"><img src=x onerror=alert(1)>` for the `tooltip`.

I don't exactly remember what values I used, but it should be straightforward enough based on this code.

Reply:
Should any further changes to this/menu logic be delayed until #7474 is in?

Reply:
we could fix this error and then move on if the other one will take longer?

Reply:
Fixed this, would you mind retesting when you get the chance? Thanks

Reply:
@daniel-beck  / @jenkinsci/core-security-review do you mind re-checking this please?

Reply:
Still broken per https://github.com/jenkinsci/jenkins/pull/7084#pullrequestreview-1360970874 because my advice was ignored.

Reply:
Good spot, I've updated it using xmlEscape from 'security/util.js'. Let me know if that's okay with you, the scenario you outlined below no longer occurs.